src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -112,6 +113,25 @@ public string NamedUser
             }
         }
 
+        private InboxHandler onMessageCenterUpdated;
+        public event InboxHandler OnMessageCenterUpdated
```

Comment:
Try rewriting it like this

```
private addedInboxListener = false;
private EventHandler onMessageCenterUpdated;
public event EventHandler OnMessageCenterUpdated
{
    add
    {
        onMessageCenterUpdated += value;
        if (!addedInboxListener) {
            MessageCenterClass.Shared().Inbox.AddListener(this);
            addedInboxListener = true;
        }
    }

    remove
    {
        onMessageCenterUpdated -= value;
        if (onMessageCenterUpdated == null)
        {
            MessageCenterClass.Shared().Inbox.RemoveListener(this);
            addedInboxListener = false;
        }
    }
}
```

Then you don't need to define InboxHandler anywhere, and you can invoke `onMessageCenterUpdated` like this:

```
onMessageCenterUpdated?.Invoke(this, EventArgs.Empty);
```

Reply:
I should have listened to you earlier, that so much easier with the EventHandler !! Thank you !